### PR TITLE
Feature/auction window

### PR DESCRIPTION
**Pull request by Peleccotur**
- Updating value on manual adjust
- Clearing selected bid on close
- Add confirmation window

### CODE REVIEW

File: Modules/AuctionManager/GUI.lua [hunk 1]
```diff
@@ -89,6 +89,9 @@ local function CreateBidWindow(self)
         if selected.cols == nil then return false end -- Handle column titles click
         self.awardPlayer = selected.cols[1].value or ""
         -- self.awardValue = selected.cols[4].value or 0
+        if not self.awardValue or self.awardValue == '' then
+            AuctionManagerGUI:UpdateBids(self)
```
Comment (lantisnt) [inline Modules/AuctionManager/GUI.lua:93]: Syntactic sugar: `Array:Method()` is same as `Array.Method(self)` thus we don't pass `self` when using `:`

File: Modules/AuctionManager/GUI.lua [hunk 2]
```diff
@@ -272,7 +275,7 @@ function AuctionManagerGUI:GenerateAuctionOptions()
         award_value = {
             name = "Award value",
             type = "input",
-            set = (function(i,v) self.awardValue = tonumber(v) or 0 end),
+            set = (function(i,v) AuctionManagerGUI:setInputAwardValue(self, v) end),
```
Comment (lantisnt) [inline Modules/AuctionManager/GUI.lua:278]: Syntactic sugar: `Array:Method()` is same as `Array.Method(self)` thus we don't pass `self` when using `:`

File: Modules/AuctionManager/GUI.lua [hunk 3]
```diff
@@ -289,6 +292,11 @@ function AuctionManagerGUI:GenerateAuctionOptions()
                 self.awardPlayer = ""
                 self:Refresh()
             end),
+            confirm = (function()
+                return "Are you sure, you want to award " .. self.itemLink ..
```
Comment (lantisnt) [inline Modules/AuctionManager/GUI.lua:296]: Lets use `string.format`. There should be one as an example in the `AuctionManager:Award`

File: Modules/AuctionManager/GUI.lua [hunk 4]
```diff
@@ -319,6 +327,9 @@ function AuctionManagerGUI:Create()
     f:AddChild(CreateOptions(self))
     f:AddChild(CreateBidWindow(self))
 
+    -- Clear active bid on close
+    f:SetCallback('OnClose', function() AuctionManagerGUI:ClearSelectedBid(self) end)
```
Comment (lantisnt) [inline Modules/AuctionManager/GUI.lua:331]: I don't think we should do this. If someone accidentaly closes the window the bids will clear during the auction.

File: Modules/AuctionManager/GUI.lua [hunk 5]
```diff
@@ -347,7 +358,7 @@ local function GetTopBids()
     return max, second
 end
 
-local function UpdateAwardValue(self)
+function AuctionManagerGUI:UpdateAwardValue(self)
```
Comment (lantisnt) [inline Modules/AuctionManager/GUI.lua:361]: Syntactic sugar

File: Modules/AuctionManager/GUI.lua [hunk 6]
```diff
@@ -362,9 +373,23 @@ local function UpdateAwardValue(self)
     end
 end
 
+function AuctionManagerGUI:setInputAwardValue(self, v)
```
Comment (lantisnt) [inline Modules/AuctionManager/GUI.lua:376]: Syntactic sugar

File: Modules/AuctionManager/GUI.lua [hunk 7]
```diff
@@ -362,9 +373,23 @@ local function UpdateAwardValue(self)
...
+    self.awardValue = tonumber(v) or 0;
+    self.top:SetStatusText("Awarding to " .. self.awardPlayer .. " for " .. self.awardValue)
+    self:Refresh()
+end
+
+function AuctionManagerGUI:ClearSelectedBid(self)
```
Comment (lantisnt) [inline Modules/AuctionManager/GUI.lua:382]: Syntactic sugar

File: Modules/AuctionManager/GUI.lua [hunk 8]
```diff
@@ -362,9 +373,23 @@ local function UpdateAwardValue(self)
...
+    self.awardValue = ""
+    self.awardPlayer = ""
+    self.top:SetStatusText("")
+    self.st:ClearSelection()
+end
+
 function AuctionManagerGUI:UpdateBids()
     LOG:Trace("AuctionManagerGUI:UpdateBids()")
-    UpdateAwardValue(self)
+    AuctionManagerGUI:UpdateAwardValue(self)
```
Comment (lantisnt) [inline Modules/AuctionManager/GUI.lua:392]: Syntactic sugar

File: Modules/AuctionManager/GUI.lua [hunk 9]
```diff
@@ -398,6 +423,8 @@ function AuctionManagerGUI:Toggle()
     LOG:Trace("AuctionManagerGUI:Toggle()")
     if not self._initialized then return end
     if self.top.frame:IsVisible() or not ACL:IsTrusted() then
+        -- Award reset on closing BidWindow.
+        AuctionManagerGUI:ClearSelectedBid(self)
```
Comment (lantisnt) [inline Modules/AuctionManager/GUI.lua:427]: Syntactic sugar

**Review by lantisnt — CHANGES_REQUESTED** [review]

**Comment by lantisnt** [comment]
Generally looks good, jus with some syntactic sugar to fix 👍.
I'll also add luacheck for the PRs as it looks like the action didn't run.

**Review by Peleccotur — COMMENTED** [review]
Updated to string.format for confirm box

File: Modules/AuctionManager/GUI.lua [hunk 10]
```diff
@@ -362,9 +376,23 @@ local function UpdateAwardValue(self)
     end
 end
 
+function AuctionManagerGUI:setInputAwardValue(v)
+    self.awardValue = tonumber(v) or 0;
+    self.top:SetStatusText("Awarding to " .. self.awardPlayer .. " for " .. self.awardValue)
+    self:Refresh()
```
Comment (lantisnt) [inline Modules/AuctionManager/GUI.lua:382]: Do we need this refresh here?

Comment (Peleccotur) [reply Modules/AuctionManager/GUI.lua:382]: I am not sure atm. I had issues with status text not updating earlier, but refreshing the whole gui seems a bit overkill. Can test again later

**Pull request merged**